numpy/add_newdocs.py

```diff
@@ -2926,10 +2926,14 @@ def luf(lamdaexpr, *args, **kwargs):
     """
     Tuple of array dimensions.
 
-    Notes
-    -----
-    May be used to "reshape" the array, as long as this would not
-    require a change in the total number of elements
+    The shape property is usually used to get the current shape of an array,
+    but may also be used to reshape the array in-place by assigning a tuple of
+    array dimensions to it.  As with `numpy.reshape`, one of the new shape
+    dimensions can be -1, in which case its value is inferred from the length
+    of the array and remaining dimensions. Reshaping an array in-place will
+    fail if a copy is required, and it affects any code holding a reference to
```

Comment:
"affects any code holding a reference to the array in an indirect way" almost suggests that it affects views of the same array data, which is not the case.

So I'm not sure this part is helpful. It's really a statement about how creating new objects should be preferred to mutation (which I generally agree with), but this sort of general coding advice feels slightly out of place in a doc-string.

Reply:
I wanted to discourage using it for reshaping. Long ago I was tempted make a PR making it readonly, but backwards compatibility was a problem. Then again, I would rather `ndarray.reshape` raised an error if a copy was needed. Oh well ;)

Hmm. how come `owndata` if false here when a copy is made?
```
In [18]: ones((4,2))[::2].reshape(4).flags
Out[18]: 
  C_CONTIGUOUS : True
  F_CONTIGUOUS : True
  OWNDATA : False
  WRITEABLE : True
  ALIGNED : True
  UPDATEIFCOPY : False
```

Reply:
Removed the comment about the mutation.

Reply:
> Then again, I would rather ndarray.reshape raised an error if a copy was needed.

Exactly. This is one valid reason for mutating `.shape`. See #9818 for my proposal for adding a `copy` argument to `reshape()`.